src/arch/linux/mapping/mapping.c

```diff
@@ -307,14 +316,35 @@ int alias_mapping(int cap, dosaddr_t targ, size_t mapsize, int protect, void *so
     if (config.cpu_vm == CPUVM_VM86)
       mem_bases[VM86_BASE] = 0;
 #endif
+    /* intermediate dirty-logged VGA memory area for KVM:
+     * KVM physical 0xa0000 -> KVM_VGA_BASE -alias> vgaemu memory
+     * This area is always writable, mprotect not applied at this level
+     * (but applied for KVM page tables with instremu only)
+     */
+    if (is_kvm_map(cap)) {
+      const int prot = PROT_READ|PROT_WRITE|PROT_EXEC;
+      unsigned char *kvm_vga_base = mmap_mapping(MAPPING_SCRATCH|MAPPING_KVM,
+	(void *)-1, VMEM_SIZE, prot);
+      int offs = config.umb_a0 ? GRAPH_SIZE : 0;
+      if (config.umb_b0) {
+	if (!config.umb_a0) // UMB only at b0000 -> two regions
```

Comment:
This is probably not the best place for
such calcs. Can they be moved to the
caller?
Also I really wonder why not to just add
a full KVM_BASE - it will likely not require
all the alike kludges because UMBs are
properly mapped for other bases, so they
would be for kvm too?

Also currently mmap_kvm() is never called
from mapping.c because we map behind
kvm's back. Only mprotect_kvm() is legally
called from mapping.c. I think if KVM_BASE
is added, this all will remain so?

But at a very minimum, please move that
code to the caller, which is low_mem_init()
I guess.

Reply:
I think KVM_BASE suits well.
If for now it would only mean to bypass
mprotect() for that particular region,
then it's still a very small patch.
But in the future MEM_BASE can
have vga region unaliased (for memcmp)
and KVM_BASE would have it still
aliased.

Reply:
Or for example MEM_BASE would have
the cpu-emu protection where KVM_BASE
doesn't have such.
Why not to try KVM_BASE? :)